src/sort.jl

```diff
@@ -0,0 +1,102 @@
+import Base.@_inline_meta
+import Base.Order: Ordering, Forward, ReverseOrdering, ord
+import Base.Sort: Algorithm, defalg, lt, sort
+
+
+struct BitonicSortAlg <: Algorithm end
+struct MinSizeSortAlg <: Algorithm end
+struct MinDepthSortAlg <: Algorithm end
+const MinSortAlg = Union{MinSizeSortAlg,MinDepthSortAlg}
+
+const BitonicSort = BitonicSortAlg()
+const MinSizeSort = MinSizeSortAlg()
+const MinDepthSort = MinDepthSortAlg()
+
+defalg(::SVector) = BitonicSort
+
+function sort(a::SVector;
+              alg::Algorithm = defalg(a),
+              lt = isless,
+              by = identity,
+              rev::Union{Bool,Nothing} = nothing,
+              order::Ordering = Forward)
+    ordr = ord(lt, by, rev, order)
+    println(Size(a))
+    (Size(a) == Size(0) || Size(a) == Size(1)) && return a
+    return _sort(Size(a), alg, ordr, a)
+end
+
+_sort(::Size{T}, alg, _, _) where T =
```

Comment:
Is this breaking functionality that might have previosly worked? Can we bail out of immutable algorithms and use a mutable approach (via e.g. `copymutable` and `sort!`) when we don't know the algorithm?